draft-barnes-mls-protocol.md

```diff
@@ -1435,44 +1434,219 @@ all arrive at the following state:
-because the receiving party can choose to ignore it (if the value is not known, or if
-transcript security is not required).
+The Group identifier and epoch allows a device to know which Group secret
+should be used and from which Epoch secret to start computing other secrets
+and keys. The participant identifier is used to derive the participant
+Application secret chain from the initial shared Application secret.
+The application message-counter field is used to determine which Application
+secret should be used from the chain to compute the correct AEAD keys
+before performing decryption.
+
+The signature field, allows to privide strong authentication of the
+plaintext and avoid Group participants to impersonate other participants.
+Note, that this non-repudiability property does not necessarily contradict
+deniability (see {{authentication}}).
+
+[[ OPEN ISSUE: Should the padding be required for Handshake messages ?
+Can an adversary get more that the position of a participant in the tree
+without padding ? Should the base ciphertext block length be negotiated or
+is is reasonnable to allow leaking an interval for the length of the plaintext
+by allowing to send a variable number of ciphertext blocks ? ]]
+
+Padding of Application messages SHOULD be enforced to provide some resistance
+against traffic analysis techniques over encrypted traffic.
+{{?CLINIC=DOI.10.1007/978-3-319-08506-7_8}}
+{{?HCJ16=DOI.10.1186/s13635-016-0030-7}}
+While MLS might be less suceptible to serve the same payload multiple time across
+a lot of ciphertexts than traditionnal web servers, it might still provide
+the attacker enough information to mount an attack. If Alice asks Bob:
+"When are we going to the movie ?" the answer "Wednesday" might be leaked
+by the ciphertext length to an adversary expecting Alice to provide Bob
+with a day of the week at some point in their discussion.
+
+Similarily to TLS 1.3, the MLS messages MUST be padded before AEAD encryption
+with zero-valued bytes. Upon AEAD decryption, the length field of the plaintext
+is used to compute the number of bytes to be removed from the plaintext to get the
+correct data.
+If the padding mechanism is used to improve protection against timing side-channels,
+it MUST be implemented in a "constant-time" at the MLS layer and above.
+
+### Delayed and Reordered Application messages
+
+Since each MLSCiphertext contains the Group identifier, the epoch and a
+message counter, a participant receiving Application messages out of order
+is able to compute the correct AEAD decryption keys if he kept the Application
+secret long enough.
+
+For usability, MLS might require to keep the Application secrets for a
+certain amount of time to retain the ability to decrypt delayed or out of
+order messages, possibly still in transit while a decryption is being done.
+Note that keeping these secrets will considerably weaken the cryptographic
+security guarantees expected at the protocol level.
```

Comment:
This is pretty vague. At least leave a placeholder for a real discussion.

Reply:
Added a discussion item.